source/posix/process.c

```diff
@@ -14,10 +14,48 @@
  */
 
 #include <aws/common/process.h>
+#include <aws/common/string.h>
 
+#include <ctype.h>
+#include <stdio.h>
 #include <sys/types.h>
 #include <unistd.h>
+#define MAX_BUFFER_SIZE (2048)
 
 int aws_get_pid(void) {
     return (int)getpid();
 }
+
+static bool s_isspace(uint8_t c) {
+    return isspace((int)c) != 0;
+}
+
+struct aws_string *aws_get_command_output(struct aws_allocator *allocator, const char *command) {
```

Comment:
The only difference between the two implementations looks like popen vs. _popen and pclose vs. _pclose.  Seems refactorable such that about 98% of the code is shared, with a pair of function pointers being the only platform specific parts.

Reply:
Thought about this, but if later we introduce more functionalities, and if those functions have bigger differences platform wide, we may have to put them in separate files without shared block. Then this function and those (future) functions are going to be here and there.